.github/proverCiScripts/benchmarks_result_reporting/reporting_modules.py

```diff
@@ -0,0 +1,212 @@
+            logdata = {
+                    'pull_request': pull_request,
+                    'circuit'     : circuit,
+                    'degree'      : degree,
+                    'result'      : result,
+                    'setup_gen'   : sg,
+                    'proof_gen'   : pg,
+                    'proof_ver'   : pv
+                }
+        else:
+            result = 'FAILED'
+            logdata = {
+                'pull_request': pull_request,
+                'circuit'     : circuit,
+                'degree'      : degree,
+                'result'      : result,
+                'setup_gen'   : 'None',
+                'proof_gen'   : 'None',
+                'proof_ver'   : 'None'
+            }        
+
+    else:
+        result = 'None'
+        logdata = {
+            'pull_request': pull_request,
+            'circuit'     : circuit,
+            'degree'      : degree,
+            'result'      : result,
+            'setup_gen'   : 'NoTestExecuted',
+            'proof_gen'   : 'NoTestExecuted',
+            'proof_ver'   : 'NoTestExecuted'
+            }
+
+    
+    return logdata
+
+def load_stats(file):
+    '''
+    loads raw mem/cpu sar data from csv to dataframe
+    '''
+    try:
+        with open(file,'r') as filedata:
+            filedatalist = [i for i in filedata.read().splitlines()]
+            header = [i for i in filedatalist if 'LINUX-RESTART' in i][0]
+            cpus = header.split('(')[1].split()[0]
+            cpudatalist = [i for i in filedatalist if 'LINUX-RESTART' not in i]
+            columns = cpudatalist[0].split(';')
+            cpudatalist = [i for i in cpudatalist if 'hostname' not in i]
+            df = pd.DataFrame([i.split(';') for i in cpudatalist], columns=columns)
+        return df, cpus
+    except Exception as e:
+        print(e)
+        return None
+
+def process_cpustats(statsdf):
+    '''
+    accepts cpu stats raw data from csv and returns a dataframe for further processing
+    '''
+    statsdf = statsdf[['timestamp', '%idle']]
+    # statsdf = statsdf.set_index('timestamp')
+    statsdf['%idle']   = pd.to_numeric(statsdf['%idle'])
+    statsdf['utilizationall'] = statsdf['%idle'].apply(lambda x:round(float(100) - x, 2 ))
+    statsdf = statsdf[['timestamp','utilizationall']]
+    cpu_all_Max = statsdf['utilizationall'].max()
+    cpu_all_Average = statsdf['utilizationall'].mean()
+    return statsdf, cpu_all_Max,cpu_all_Average
+
+
+def process_memstats(df):
+    '''
+    accepts ram stats raw data  and returns a dataframe for further processing
+    '''
+    statsdf = df[['timestamp', 'kbmemused']]
+    # statsdf = statsdf.set_index('timestamp')
+    statsdf['kbmemused']   = pd.to_numeric(statsdf['kbmemused'])
+    statsdf['utilizationgb']   = statsdf['kbmemused'].apply(lambda x: round(x/float(1000000),2))
+    statsdf = statsdf[['timestamp','utilizationgb']]
+    max_ram = statsdf['utilizationgb'].max()
+    return statsdf, max_ram
```

Comment:
ditto

Reply:
ends with new line but only shown in edit mode :)